### ISSUE TITLE
Markdown bug: punctuation marks break italics formatted with underscores

### ISSUE THREAD

**Issue by DavidNemeskey** [jupyter/notebook#4908]
In GitHub markdown, and in a previous version of Jupyter (don't remember which, sorry), it was possible to e.g. display quotations in italics formatted with underscores "_like this_". Unfortunately, Jupyter 1.0.0 displays this as "\_like this\_". The other formatting options work:

- "*asterisk for italics*"
- "__underscore for bold__"
- "**asterisk for bold**"

The error manifests not just with quotation marks, but with other punctuation marks (comma, dot, quastion mark, etc.) as well.

**Comment by laowantong** [comment]
Same problem with html markup, e.g. `<u>_foobar_</u>` is not working anymore, although `<u>*foobar*</u>` still renders as intended.

**Comment by laowantong** [comment]
The problem is triggered by the _right_ symbol only, i.e., when the closing underscore is _immediately followed_ by any symbol among `!"#%&'()*+,-/:;<=>?@[]^{}` (at least). These symbols are (wrongly) treated as alphanumeric. Note that the dot `.` constitutes an exception, i.e., `_FOOBAR_.` is correctly italicized.

I would then change the title of the issue into something like: "Markdown bug: italics formatted with underscores are broken when followed by a non-alphanumeric symbol".

To reproduce the problem, execute the following python code in a cell, and copy the result in a Markdown cell:

```python
print("| symbol | left | right | both |")
print("|--|--|--|--|")
for c in """!"#%&'()*+,-./:;<=>?@[]^{}""":
    print(f"| {c}", end=" | ")
    print(f"{c}_FOOBAR_", end=" | ")
    print(f"_FOOBAR_{c}", end=" | ")
    print(f"{c}_FOOBAR_{c} |")
```

![screenshot](https://user-images.githubusercontent.com/3048819/67013031-ba4ef780-f0f2-11e9-89ef-8066c42997d8.png)

**Comment by DavidNemeskey** [comment]
Good catch! If you wish to change the title, go ahead, though I am not sure we need to cramp every detail into it. :smile:

**Comment by laowantong** [comment]
Thanks! I have not the rights to edit the title, which IMHO is already quite adequate (I have found the issue with the search keywords Markdown + italics).

**Comment by laowantong** [comment]
In the meantime, here is a regular expression to replace (hopefully) every broken formatting by the asterisked form. Warning: control the substitutions before committing, as I don't test whether the searched string appears in a Markdown cell or a code cell.

- Find: `\b_([^\n_]+?)_\b(?![  \\\.\]])`
- Replace: `*$1*`

Tested under Visual Studio Code:
![image](https://user-images.githubusercontent.com/3048819/67020803-0142ea00-f0ff-11e9-891b-4dc12592ee65.png)
